vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/descriptor/scalar/BooleanDescriptor.kt

```diff
@@ -31,4 +31,13 @@ data class BooleanDescriptor(
      * @return [List] of [Attribute]
      */
     override fun layout(): List<Attribute> = SCHEMA
+    /**
+     * Returns a copy of this [BooleanDescriptor] with new [RetrievableId] and/or [DescriptorId]
+     *
+     * @param id [DescriptorId] of the new [BooleanDescriptor].
+     * @param retrievableId [RetrievableId] of the new [BooleanDescriptor].
+     * @param field [Schema.Field] the new [BooleanDescriptor] belongs to.
+     * @return Copy of this [BooleanDescriptor].
+     */
+    override fun copy(id: DescriptorId, retrievableId: RetrievableId?, field: Schema.Field<*, BooleanDescriptor>?) = BooleanDescriptor(id, retrievableId, this.value, field)
```

Comment:
Are the default values from the super type inherited here or do they need to be added explicitly?

Reply:
They are inherited.